packages/ui/src/components/UnitInput/index.tsx

```diff
@@ -0,0 +1,329 @@
+  'data-size': 'small' | 'medium' | 'large'
+  'data-focus': boolean
+  'data-success': boolean
+  'data-error': boolean
+  'data-disabled': boolean
+  'data-readonly': boolean
+}>`
+  border: 1px solid ${({ theme }) => theme.colors.neutral.border};
+  border-radius: ${({ theme }) => theme.radii.default};
+
+  &:hover,
+  &:focus {
+    text-decoration: none;
+    border-color: ${({ theme }) => theme.colors.primary.border};
+    box-shadow: none;
+  }
+
+  &[data-readonly='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundWeak};
+    border-color: ${({ theme }) => theme.colors.neutral.border};
+    cursor: default;
+  }
+  &[data-disabled='true'] {
+    background: ${({ theme }) => theme.colors.neutral.backgroundDisabled};
+    border-color: ${({ theme }) => theme.colors.neutral.borderDisabled};
+    cursor: not-allowed;
+  }
+
+  &[data-size='small'] {
+    height: ${INPUT_SIZE_HEIGHT['small']}px;
+  }
+  &[data-size='medium'] {
+    height: ${INPUT_SIZE_HEIGHT['medium']}px;
+  }
+  &[data-size='large'] {
+    height: ${INPUT_SIZE_HEIGHT['large']}px;
+  }
+  &[data-focus='true'] {
+    box-shadow: ${({ theme }) => theme.shadows.focusPrimary};
+    border-color: ${({ theme }) => theme.colors.primary.borderHover};
+    & > ${StyledInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.primary.border};
+    }
+  }
+  &[data-success='true'] {
+    border: 1px solid ${({ theme }) => theme.colors.success.border};
+    & > ${StyledInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.success.border};
+    }
+  }
+  &[data-error='true'] {
+    border: 1px solid ${({ theme }) => theme.colors.danger.border};
+    & > ${StyledInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.danger.border};
+    }
+    & > ${StyledInputWrapper}:hover {
+      border-right-color: ${({ theme }) => theme.colors.danger.border};
+    }
+  }
+  &[data-error='true'] {
+    border: 1px solid ${({ theme }) => theme.colors.danger.border};
+    & > ${StyledInputWrapper} {
+      border-right-color: ${({ theme }) => theme.colors.danger.border};
+    }
+  }
+`
+
+type UnitInputValue = { inputValue: number; unit: string }
+
+type UnitInputProps = {
+  id?: string
+  className?: string
+  name: string
+  disabled?: boolean
+  maxValue?: number
+  minValue?: number
+  value?: UnitInputValue['inputValue']
+  unitValue?: UnitInputValue['unit']
+  onChange: (value: UnitInputValue['inputValue']) => void
+  onChangeUnitValue: (values: string[]) => void
+  options: OptionType[]
+  placeholder?: string
+  selectInputWidth?: number
+  size?: 'small' | 'medium' | 'large'
+  'data-testid'?: string
+  notice?: string
+  required?: boolean
+  helper?: string
+  unitError?: string
+  width?: ComponentProps<typeof Stack>['width']
+  placeholderUnit?: string
+  autoFocus?: boolean
+  error?: boolean
+  success?: boolean
+  label?: string
+  labelInformation?: ReactNode
+  step?: number | string
+  readOnly?: boolean
+}
```

Comment:
For the following props: name, id, disabled, placeholder, autoFocus, required, and if you find more. You can do:

```ts
} & InputHTMLAttributes<HTMLInputElement>,
  | 'name'
  | 'id'
  | 'placeholder'
  | ...
```